### PR TITLE
Moved TDT to Python

### PR DESCRIPTION
**Pull request by tpoterba**
Still relies on TGenotype. I'm curious what @cseed has planned to move everything over.

### CODE REVIEW

**Comment by tpoterba** [comment]
Re-assigned to Amanda to balance the load from Patrick.

File: python/hail/dataset.py [hunk 1]
```diff
@@ -5746,17 +5813,18 @@ def trio_matrix(self, pedigree, complete_trios=False):
         are the sample IDs of the trio probands. The column annotations and
         entries of the matrix are changed in the following ways:
 
-        The new column annotation schema is a ``Struct`` with three ``Struct`` children
-        (``proband``, ``father``, and ``mother``),
-        each with an ``id`` and ``annotations`` field. The schema of each ``annotations``
-        field is the column annotation schema of the input dataset.
+        The new column annotation schema is a ``Struct`` with five children
+        (structs ``proband``, ``father``, and ``mother``, boolean ``sex``, and string ``famID``).
```
Comment (catoverdrive) [inline python/hail/dataset.py]: I think ``sex`` here is ``isFemale``, right?

File: python/hail/dataset.py [hunk 2]
```diff
@@ -4980,38 +4987,98 @@ def tdt(self, pedigree, root='va.tdt'):
...
+                            mom: g.mother.nNonRefAlleles(),
+                            ploidy: ploidy}}
+                        )[{index}])
+                .sum()'''
+
+        per_variant_results = (
+            trio_matrix
+                .annotate_variants_expr([s.format(name='t', index=0), s.format(name='u', index=1)])
+                .variants_table()
+                .annotate('transmitted = va.t, untransmitted = va.u')
```
Comment (catoverdrive) [inline python/hail/dataset.py]: are these supposed to correspond to 'nTransmitted' and 'nUntransmitted' in the docs?

File: python/hail/dataset.py [hunk 3]
```diff
@@ -4980,38 +4987,98 @@ def tdt(self, pedigree, root='va.tdt'):
...
+        trio_matrix = trio_matrix.filter_samples_expr('isDefined(sa.isFemale)')
+
+        hom_ref = 0
+        het = 1
+        hom_var = 2
+
+        auto = 2
+        hemiX = 1
+
+        t = TDict(TStruct(['kid', 'dad', 'mom', 'ploidy'], [TInt32(), TInt32(), TInt32(), TInt32()]), TArray(TInt32()))
```
Comment (catoverdrive) [inline python/hail/dataset.py]: you mentioned that 'ploidy' might not be the right term here; I think the original uses 'copyState'?

File: python/hail/dataset.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/dataset.py]: good catch.

**Pull request merged**